src/main/kotlin/com/sourcegraph/cody/auth/ui/LoadingAccountsDetailsProvider.kt

```diff
@@ -41,9 +42,13 @@ abstract class LoadingAccountsDetailsProvider<in A : Account, D : AccountDetails
           }
           .exceptionally {
             val error = CompletableFutureUtil.extractError(it)
-            val errorMessage =
-                error.localizedMessage.takeWhile { c -> c.isLetterOrDigit() || c.isWhitespace() }
-            DetailsLoadingResult(null, null, errorMessage, false)
+            if (error.message == UNAUTHORIZED_ERROR_MESSAGE) {
+              DetailsLoadingResult(null, null, error.localizedMessage, true)
+            } else {
+              val errorMessage =
+                  error.localizedMessage.takeWhile { c -> c.isLetterOrDigit() || c.isWhitespace() }
```

Comment:
What about `$` in the class names? What exactly you try to trim there?

Reply:
This is the previous behaviour. I only improved the handling for unauthorized error.

When I switch the internet connection of I see this:
<img width="325" alt="image" src="https://github.com/sourcegraph/jetbrains/assets/19799111/c857a7af-5eab-49f1-8d4b-3dc2af84dfc9">

The code is supposed to extract the root cause of the problem (but it does not work properly in many cases).